### ISSUE TITLE
Does not compile with Delphi 7

### ISSUE THREAD

**Issue by dummzeuch** [MahdiSafsafi/DDetours#61]
Even though you claim that it is compatible with Delphi 7, it doesn't compile:

[Error] DDetours.pas(1387): Undeclared identfier 'PNativeInt'

(Or did I take the wrong source code? I took the master branch from
https://github.com/MahdiSafsafi/DDetours.git )

**Comment by dummzeuch** [comment]
The fix is easy:

add the following to the ddetours interface section:

{$IF (CompilerVersion >= 15) and (CompilerVersion <= 18.5}
// Delphi 7 to 2007: NativeInt is wrong and PNativeInt doesn't exist
type
  NativeInt=Integer;
  PNativeInt=^NativeInt;
{$IFEND}

Note that the declaration of NativeInt in Delphi 7 to 2007:

https://blog.dummzeuch.de/2018/09/08/nativeint-nativeuint-type-in-various-delphi-versions/

**Comment by dummzeuch** [comment]
Another problem is the UTF-8 BOM in CPUID.pas. I had to delete it to make it compile with Delphi 7

**Comment by dummzeuch** [comment]
And finally I had to remove the namespace prefixes in uMain.pas (Demo1) and change the LPCWSTR in the TMessageBox type and the InterceptMessageBox function to LPCSTR:

type
  TMessageBox = function(hWnd: hWnd; lpText, lpCaption: LPCSTR; uType: UINT): Integer; stdcall;

function InterceptMessageBox(hWnd: hWnd; lpText, lpCaption: LPCSTR; uType: UINT): Integer; stdcall;

After these changes everyting compiled and the demo worked in Delphi 7.

I think this change will also still compile and work with newer Delphi versions, but I haven't tested it.

**Comment by MahdiSafsafi** [comment]
Sorry for that. It was working perfectly on D7 the moment when I announced the immediate availability of DDetours v2.2. Later I introduced PNativeInt and I forgot to check on D7.

Give some moment, I'll make a patch based on your recommendation to fix all those incompatibility issues.
 
N.B: On my machine, I only have D7, Lazarus and Rio. So I cant know for sure about other versions but if it works on D7 it should for all.

Thanks.
